pimd/pim_igmp_mtrace.c

```diff
@@ -373,7 +373,9 @@ static int mtrace_un_forward_packet(struct pim_instance *pim, struct ip *ip_hdr,
 	if (fd < 0)
 		return -1;
 
+#if PIM_IPV == 4
```

Comment:
Any reason for adding this IPV4 check? For IPV6, uninitialized fd has been used for socket-bind for e.g. I think, since Mtrace is implemented only for IGMP case, we can remove this check and enhance this function later, when we support Mtrace for MLD

Reply:
I have added this as pim_ip_socket_hdr() is used only for IPv4. This is just a work around for now. Ideally these files/functions would be compiled only for IPv4